src/engine/xrinput.js

```diff
@@ -1,12 +1,36 @@
 import State from "./state";
 import Renderer from "./renderer";
-import { XRControllerModelFactory } from "three/examples/jsm/webxr/XRControllerModelFactory.js";
 
 class XRInputClass {
   constructor() {
+    /** Deprecated: Use leftControllerGrip or rightControllerGrip instead */
     this.controllerGrips = [];
     this.inputSources = null;
-    this.controllerModelFactory = new XRControllerModelFactory();
+
+    this._leftController = {};
```

Comment:
What happened to wrapping this in a controller class?

Reply:
So the main reason I went this route is so that we can simply assign the controller Group that the renderer returns to this property, without having to do any sort of copying of values. The controller data is a generic Group, but it would probably be better for the Sandcastle engine if it were a class with specific properties. That would make the documentation clearer for a user of the engine as well.

So the benefit right now of doing it this way is that we don't have to continuously copy over values. I do not think this will be the best long term strategy though.